### DISCUSSION TITLE
Custom Cell

### DISCUSSION THREAD

**Discussion by antoniobeltran** [glideapps/glide-data-grid#606] [Q&A] [answered]
I see there is an option for custom cells, but the custom cells appear to be something like a date picker which is imported from glide-data-grid-cells. I wouldn't call that custom though, right? `kind: GridCellKind.Custom` as it's something glide date grid has.

I was trying to make my own date picker that is a react component to return so I can use an existing react date picker that we already have, hence the reason why I was looking at the custom story book, but it appears when I use `GridCellKind.Custom` I'm not making my own custom cell, but using one of the cells glide-data-grid has. Can I make my own custom date picker that's a react component or do I have to do canvas? I don' t have much experience with canvas. Can you make canvas return a react component? Was looking at [this](https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-dataeditor-demos--draw-custom-cells)

**Comment by jassmith** [comment, marked as answer]
You can build entirely custom cells using the exact same API's the `glide-data-grid-cells` package uses. That package is built entirely using public API and was intended to be both useful on its own and as a demo.

**Comment by antoniobeltran** [comment]
ok, that makes sense, and that is what it's called custom cells because
glide-data-grid-cells is making custom cells. So I need to dig into
how glide-data-grid-cells
works

On Wed, Jan 18, 2023 at 11:08 AM Jason Smith ***@***.***>
wrote:

> You can build entirely custom cells using the exact same API's the
> glide-data-grid-cells package uses. That package is built entirely using
> public API.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glideapps/glide-data-grid/discussions/606#discussioncomment-4719007>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJL3R5FYIIQC3RIUI5AI7GDWTAIPPANCNFSM6AAAAAAT7IK5KQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
<https://about.me/atno?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb>
Antonio Beltran
about.me/atno
<https://about.me/atno?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb>
